src/Service/IncentiveService.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use App\Audit\Log;
+
+class IncentiveService
+{
+    protected $rdrApiService;
+    protected $siteService;
+    protected $userService;
+    protected $em;
+    protected $loggerService;
+
+    public function __construct(
+        RdrApiService $rdrApiService,
+        SiteService $siteService,
+        UserService $userService,
+        EntityManagerInterface $em,
+        LoggerService $loggerService
+    ) {
+        $this->rdrApiService = $rdrApiService;
+        $this->siteService = $siteService;
+        $this->userService = $userService;
+        $this->em = $em;
+        $this->loggerService = $loggerService;
+    }
+
+    public function getRdrObject($incentive, $type = 'create')
```

Comment:
Instead of passing around the strings 'create', 'amend', and 'cancel', can you use class constants?

Reply:
Can you add tests for this method?